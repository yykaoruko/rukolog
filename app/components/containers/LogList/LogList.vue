<template>
  <div class="log-list">
    <template v-for="(item, index) in logList">
      <LogCardLink :key="index" v-bind="logCardLinkProps(item)" />
    </template>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import LogCardLink from '@/components/combinations/LogCardLink/LogCardLink.vue';
import { logDate as logDateReg } from '@/constants/regex';
const jsonData = require('@/contents/dist/summary.json');
type LogItem = {
  base: string;
  created_at: string;
  dir: string;
  ext: string;
  sourceBase: string;
  sourceExt: string;
  tag: string[] | null;
  title: string;
};
type LogCardLinkPropsType = {
  datetime: string;
  description: string;
  to: string;
  title: string;
  tags: string[] | [];
};
export default Vue.extend({
  name: 'LogList',
  components: {
    LogCardLink,
  },
  data() {
    return {
      logList: jsonData.fileMap,
    };
  },
  methods: {
    logCardLinkProps(item: LogItem): LogCardLinkPropsType {
      const datetime = this.getDatetime(item.base);
      return {
        datetime: datetime,
        // TODO: description
        description: '',
        to: `/logs/${datetime}`,
        title: item.title,
        tags: item.tag === null ? [] : item.tag,
      };
    },
    getDatetime(value: string): string {
      const match = value.match(logDateReg);
      if (match !== null) {
        return match[0];
      } else {
        throw 'yabaiyo';
      }
    },
  },
});
</script>

<style lang="scss" scoped>
.log-list {
  color: inherit;
}
</style>
