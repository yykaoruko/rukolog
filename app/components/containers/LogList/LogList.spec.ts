import Vue from 'vue';
import { shallowMount, Wrapper } from '@vue/test-utils';
import LogList from './LogList.vue';

describe('LogList', () => {
  describe('mount', () => {
    let wrapper: Wrapper<Vue>;

    beforeEach(() => {
      wrapper = shallowMount(LogList, {
        propsData: {},
      });
    });

    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true);
    });

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
