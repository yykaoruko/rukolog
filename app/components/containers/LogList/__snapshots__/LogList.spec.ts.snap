// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogList mount snapshot 1`] = `
<div
  class="log-list"
>
  <logcardlink-stub
    class="log-list__item"
    datetime="2019-09-08"
    description=""
    tags="aa,nb"
    title="今週の振り返り"
    to="/logs/2019-09-08"
  />
  <logcardlink-stub
    class="log-list__item"
    datetime="2019-09-07"
    description=""
    tags="aa,nb"
    title="今週の振り返り"
    to="/logs/2019-09-07"
  />
</div>
`;
